sql-plugin/src/main/scala/com/nvidia/spark/rapids/CostBasedOptimizer.scala

```diff
@@ -270,6 +273,62 @@ class DefaultCostModel(conf: RapidsConf) extends CostModel {
 
 }
 
+/**
+ * Estimate the number of rows that an operator will output. Note that these row counts are
+ * the aggregate across all partitions.
```

Comment:
This is not necessarily true with how the code is written.

LocalLimit and GlobalLimit are the same code, but GlobalLimit requires a single partition, while LocalLimit does not. That means that GlobalLimit is across all partitions, because there is only a single partition, but LocalLimit is per-partition.

Reply:
I updated the logic for LocalLimit to multiple the limit by the number of output partitions for consistency.